core/shared/src/main/scala/zio/Fiber.scala

```diff
@@ -472,43 +477,16 @@ object Fiber extends FiberPlatformSpecific {
      *     at ...
      * }}}
      */
-    def prettyPrintM: UIO[String] = UIO {
-      val time = System.currentTimeMillis()
-
-      val millis  = (time - fiberId.startTimeMillis)
-      val seconds = millis / 1000L
-      val minutes = seconds / 60L
-      val hours   = minutes / 60L
-
-      val name = fiberName.fold("")(name => "\"" + name + "\" ")
-      val lifeMsg = (if (hours == 0) "" else s"${hours}h") +
-        (if (hours == 0 && minutes == 0) "" else s"${minutes}m") +
-        (if (hours == 0 && minutes == 0 && seconds == 0) "" else s"${seconds}s") +
-        (s"${millis}ms")
-      val waitMsg = status match {
-        case Suspended(_, _, _, blockingOn, _) =>
-          if (blockingOn.nonEmpty)
-            "waiting on " + blockingOn.map(id => s"#${id.seqNumber}").mkString(", ")
-          else ""
-        case _ => ""
-      }
-      val statMsg = status match {
-        case Done         => "Done"
-        case Finishing(b) => "Finishing(" + (if (b) "interrupting" else "") + ")"
-        case Running(b)   => "Running(" + (if (b) "interrupting" else "") + ")"
-        case Suspended(_, interruptible, epoch, _, asyncTrace) =>
-          val in = if (interruptible) "interruptible" else "uninterruptible"
-          val ep = s"${epoch} asyncs"
-          val as = asyncTrace.map(_.prettyPrint).mkString(" ")
-          s"Suspended(${in}, ${ep}, ${as})"
-      }
+    def prettyPrintM: UIO[String] = FiberRenderer.prettyPrintM(this)
+  }
 
-      s"""
-         |${name}#${fiberId.seqNumber} (${lifeMsg}) ${waitMsg}
-         |   Status: ${statMsg}
-         |${trace.prettyPrint}
-         |""".stripMargin
-    }
+  final case class Tree(
```

Comment:
There is a lot of overlap between this and the existing `Dump` data type. Could we just augment `Dump` to include a collection of children?

Reply:
It bugs me as well but OTOH I can see few downsides: 
1) fiber tree feels more lightweight than entire dump and ability to collect just the tree w/o messing with stacktraces feels beneficial (uninformed IMHO)
2) Data stored in `Dump` objects become (more) denormalized
3) If implemented naively - collecting subtrees few times (once for each transitive parent) have memory and performance implications. Could be important given that we usually collect such information when the application is already not in the best shape.

Reply:
What if `Dump` when to including an `Option[ZTrace]` instead of a `ZTrace`, so methods that created dumps could decide whether to capture tracing information or not? Then dumps could just be fiber trees if we wanted. I think if we changed to this representation generating a dump would just mean generating a dump for each of the parents (since the children would be included in the parent dumps) which would involve the same work as today.

Reply:
Do you mean that dump will look like that: 
Dump(fiberId: Fiber.Id, fiberName: Option[String], status: Status, children: Iterable[Dump],trace: Option[ZTrace]) and 

`def dump(fibers: Fiber.Runtime[_, _]*): UIO[Iterable[Dump]]` will return `n` dumps for `n` fibers (so become useless) and to render all stacktraces we will traverse the tree?

Reply:
Yeah, I'm thinking we would have:

```scala
final case class Dump(
  fiberId: Fiber.Id,
  fiberName: fiberName: Option[String],
  status: Status,
  children: Iterable[Dump],
  trace: Option[ZTrace]
)
```

Then `dump` would retain its signature but its meaning would change somewhat:

```scala
def dump(fibers: Fiber.Runtime[_, _]*): UIO[Iterable[Dump]]
```

Previously, this would return a "flat" structure. For example, if we did a dump of one fiber that had five children we would get an iterable that had five elements, one for each fiber. In the new encoding it would return an iterable with a single element and that element would have four children.

Then a render could be implemented as:

```scala
def render(dumps: Iterable[Dump]): String = ???
```

And the render could have access to the full structure of the fiber tree in deciding how to render it.